src/app/account/additional-components/account-flows-tab/account-flows-subtabs/account-flows-datasets-subtab/account-flows-datasets-subtab.component.ts

```diff
@@ -118,105 +125,87 @@ export class AccountFlowsDatasetsSubtabComponent extends BaseComponent implement
-                      params.datasetBasics.id,
-                      params.subscriptionId,
-                  );
-        result$.pipe(take(1)).subscribe((result: boolean) => {
-            if (result) {
-                setTimeout(() => {
-                    this.refreshNow();
-                }, this.TIMEOUT_REFRESH_FLOW);
-            }
-        });
-    }
-
     public fetchCardsData(): void {
-        this.accountFlowsCardsData$ = timer(0, environment.delay_polling_ms).pipe(
-            tap(() => {
-                this.getPageFromUrl();
-            }),
+        const polling$ = timer(0, environment.delay_polling_ms);
+        const triggerWithInitial$ = this.fetchTrigger$.pipe(startWith(undefined));
+        this.accountFlowsCardsData$ = combineLatest([polling$, triggerWithInitial$]).pipe(
             switchMap(() =>
                 this.accountService.getAccountFlowsAsCards({
                     accountName: this.accountName,
                     page: this.currentPage - 1,
-                    perPage: this.CARDS_FLOW_PROCESSES_PER_PAGE,
+                    perPage: this.processesPerPage,
                     filters: this.setFlowProcessFilters(this.accountFlowsData.datasetsFiltersMode),
                     ordering: {
                         field: this.dashboardFiltersState.selectedOrderField ?? FlowProcessOrderField.LastAttemptAt,
                         direction: this.accountFlowsFiltersService.orderDirection,
                     },
                 }),
             ),
+
+            map((result: AccountFlowProcessCardConnectionDataFragment) => {
+                let newChunkCards = result.nodes as AccountFlowProcessCard[];
+                this.hasNextPage = result.pageInfo.hasNextPage;
+
+                if (
+                    this.dashboardFiltersState.isFirstInitialization ||
+                    this.loadingCards ||
+                    (this.dashboardFiltersState.applyFilters && !this.processesCards.length)
+                ) {
+                    this.processesCards = [...this.processesCards, ...newChunkCards];
+                    newChunkCards = [];
+                }
+                if (this.isToggleCardState) {
+                    this.processesCards = newChunkCards;
+                }
+                this.isToggleCardState = false;
+                this.loadingCards = false;
+                this.accountFlowsFiltersService.updateFilters({ isFirstInitialization: false });
+                return {
+                    totalCount: result.totalCount,
+                    nodes: this.processesCards,
+                };
+            }),
         );
     }
 
+    public triggerFetch(): void {
+        this.fetchTrigger$.next();
+    }
+
+    public onScroll(): void {
+        this.processesPerPage = AppValues.UPLOAD_FLOW_PROCESSES_PER_PAGE;
+        if (this.loadingCards || !this.hasNextPage) {
+            return;
+        }
+        this.currentPage++;
```

Comment:
I think you have to recompute "currentPage" more accurately here.

Let's, say I opened the page, started scrolling down, had 3 pages loaded, and clicked "pause" on one of the cards.
My `processesPerPage` would be `3 * UPLOAD_FLOW_PROCESSES_PER_PAGE`, while `currentPage` would be 1.

Then, after pause refresh, I start scrolling down again. I expect to load page #4. You reset back the size of the page to default, but `currentPage` is not recomputed back, it was 1 in 3x page sizes, and now you just add 1 and assume normal page size. I think that would load page #2 instead of page #4.

I recommend to get rid of "page" abstraction in general, and use offsets explicitly. It would be less error-prone.